lib/backend/compat/compat.go

```diff
@@ -352,7 +353,7 @@ func ensureBlockAffinity(kvp *model.KVPair) *model.KVPair {
 		// Set AllocationBlock.HostAffinity to nil so it's never non-nil for the clients.
 		val.HostAffinity = nil
 	}
-	return &model.KVPair{Key: kvp.Key, Value: val}
+	return &model.KVPair{Key: kvp.Key, Value: val, Revision: kvp.Revision, TTL: kvp.TTL}
```

Comment:
TTL is not being set here, this is just passing the TTL value along.
If etcd has a TTL value for a key, `etcd.Get()` call will get it and pass it to the client. Now compat is just adding a layer so it should be transparent and pass TTL and Revision along with Key and Value it gets from `etcd.Get()` https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/compat/compat.go#L245